packages/fuels-contract/src/execution_script.rs

```diff
@@ -114,10 +120,181 @@ impl ExecutableFuelCall {
             .iter()
             .any(|r|
                 matches!(r, Receipt::ScriptResult { result, .. } if *result != ScriptExecutionResult::Success)
-        ) {
+            ) {
             return Err(Error::RevertTransactionError(Default::default(), receipts));
         }
 
         Ok(receipts)
     }
+
+    pub fn set_inputs(
+        &mut self,
+        inputs: Vec<Input>,
+    ) -> std::result::Result<&mut ExecutableFuelCall, Error> {
+        let mut hash_set_utxo_id = HashSet::new();
+        let mut hash_set_messages = HashSet::new();
+
+        for input in &inputs {
```

Comment:
Seems to be a few places where duplicates are being handled, you can do something like this to clean it up:
```rust
    fn extract_duplicate_ids(inputs: &[Input]) -> Vec<String> {
        inputs
            .iter()
            .filter_map(|input| {
                input
                    .utxo_id()
                    .map(Either::Left)
                    .or_else(|| input.message_id().map(Either::Right))
            })
            .duplicates()
            .map(|id| match id {
                Either::Left(id) => format!("{id:?}"),
                Either::Right(id) => format!("{id:?}"),
            })
            .collect()
    }
    
    fn validate_no_duplicate_ids(inputs: &[Input]) -> Result<(), Error> {
        let duplicate_ids = Self::extract_duplicate_ids(inputs);

        if duplicate_ids.is_empty() {
            Ok(())
        } else {
            Err(Error::InvalidData(format!(
                "Provided inputs are not unique! Duplicated ids: {duplicate_ids:?}"
            )))
        }
    }
```

then you can call `validate_no_duplicate_ids` wherever you need it, for example:
```rust
    pub fn extend_inputs(&mut self, inputs: Vec<Input>) -> Result<&mut ExecutableFuelCall, Error> {
        Self::validate_no_duplicate_ids(&inputs)?;

        self.tx.inputs_mut().extend(inputs);

        Ok(self)
    }
```
or
```rust
    pub fn set_inputs(
        &mut self,
        inputs: Vec<Input>,
    ) -> std::result::Result<&mut ExecutableFuelCall, Error> {
        Self::validate_no_duplicate_ids(&inputs)?;

        *self.tx.inputs_mut() = inputs;
        Ok(self)
    }
```

Reply:
This looks much cleaner. I'll take this but I'll have to change a few things to make it work. Thanx!